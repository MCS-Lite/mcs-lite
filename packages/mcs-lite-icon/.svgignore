<path id="Bounds" fill="none" d="M0,0h24v24H0V0z"/>
sketch:type="(.*?)"
id="(.*?)"
