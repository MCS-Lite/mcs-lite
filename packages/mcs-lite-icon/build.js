require('babel-register');
require('./src/svg-to-component');
