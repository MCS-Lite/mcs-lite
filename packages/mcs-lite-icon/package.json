{
  "name": "mcs-lite-icon",
  "version": "0.1.6",
  "description": "Convert SVG icon to React components",
  "repository": "evenchange4/mcs-lite",
  "author": "Michael Hsu <evenchange4@gmail.com>",
  "bugs": {
    "url": "https://github.com/evenchange4/mcs-lite/issues",
    "email": "evenchange4@gmail.com"
  },
  "homepage": "https://github.com/evenchange4/mcs-lite#readme",
  "files": [
    "lib"
  ],
  "main": "lib/index.js",
  "scripts": {
    "clean": "rm -rf lib",
    "prebuild": "npm run clean",
    "build": "svg-to-component 'node_modules/mcs-lite-design/lib/icon' lib",
    "createEntryFile": "exports-all 'lib/[A-Z]*.js' 'lib/index.js'",
    "postbuild": "npm run createEntryFile",
    "test": "echo 'TODO'"
  },
  "license": "ISC",
  "devDependencies": {
    "mcs-lite-design": "^0.1.3",
    "mcs-lite-scripts": "^0.1.4"
  },
  "peerDependencies": {
    "react": "^15.4.2"
  }
}
