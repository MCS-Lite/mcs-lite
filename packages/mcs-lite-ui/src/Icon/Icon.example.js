import React from 'react';
import { storiesOf } from '@kadira/storybook';
import styled from 'styled-components';
import Heading from '../Heading';
import Card from '../Card';

const mcsLiteIcon = require('mcs-lite-icon');

const Icons = Object.keys(mcsLiteIcon);

const IconWrapper = styled(Heading)`
  display: flex;
  flex-direction: column;
  align-items: center;

  > *:first-child {
    margin-bottom: 5px;
  }
`;

const StyledCard = styled(Card)`
  display: inline-flex;
  align-items: center;
  margin: 8px;
  padding: 16px;
`;

storiesOf('Icon')
  .addWithInfo(
    'API',
    '',
    () =>
      <mcsLiteIcon.IconEllipsisV />,
    { inline: true },
  )
  .addWithInfo(
    'Icon list, Custom color and size',
    '',
    () =>
      <div>
        {
          Icons.map(name => (
            <StyledCard key={name}>
              <IconWrapper color="primary" level={2}>
                {React.createElement(mcsLiteIcon[name])}
                {`<${name} />`}
              </IconWrapper>
            </StyledCard>
          ))
        }
      </div>,
    { inline: false, propTables: false },
  );
