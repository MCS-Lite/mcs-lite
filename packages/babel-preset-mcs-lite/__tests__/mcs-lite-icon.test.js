import * as babel from 'babel-core';
import mcsLitePreset from '../index';

const babelOptions = {
  presets: [mcsLitePreset],
  babelrc: false,
};

const compile = code => babel.transform(code, babelOptions).code;

it('should work with mcs-lite-icon', () => {
  expect(
    compile(
      `
import { Icon } from 'mcs-lite-icon';
console.log(Icon);
  `,
    ),
  ).toMatchSnapshot();
});

it('should work with mcs-lite-icon/lib', () => {
  expect(
    compile(
      `
import Icon from 'mcs-lite-icon/lib/Icon';
console.log(Icon);
  `,
    ),
  ).toMatchSnapshot();
});
