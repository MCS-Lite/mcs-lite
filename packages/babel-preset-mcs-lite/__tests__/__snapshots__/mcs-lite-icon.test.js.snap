// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work with mcs-lite-icon 1`] = `
"'use strict';

var _Icon2 = require('mcs-lite-icon/lib/Icon');

var _Icon3 = _interopRequireDefault(_Icon2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

console.log(_Icon3.default);"
`;

exports[`should work with mcs-lite-icon/lib 1`] = `
"'use strict';

var _Icon = require('mcs-lite-icon/lib/Icon');

var _Icon2 = _interopRequireDefault(_Icon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

console.log(_Icon2.default);"
`;
